docs/function_calling.rst

```diff
@@ -254,4 +297,26 @@ passing params with invalid, non-coercible types) or the function raises an exce
 error in a message to the model by default, allowing it up to *retry_attempts* to correct itself and retry the
 call.
 
+.. note::
+    If the model calls multiple tools in parallel, the model will be allowed a retry if *any* exception handler
+    allows it.
+
```

Comment:
Yeah this is also interesting how parallel functions play into retry logic. I'm assuming we increment retries if any of the functions fail? What happens when multiple fail (is that +1 or +n to retry count). If one of n functions fail, does the model retry all of the calls? or just the failed one. Maybe it's worth some clarification here...

Reply:
Since retries are controlled by the model anyway, the retry param is kind of fuzzy - it means more like "allow the model to retry this call if it feels like it". If one of n functions fail, the return to the model will include all n results, with the failed one noting an exception - the model can then choose to retry calling the one it failed if it feels like it. If multiple fail, that adds 1 to the retry count (think of it as number of retries allowed per round). Will clarify